CMakeLists.txt

```diff
@@ -19,72 +23,106 @@ IF ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
    3) cmake -DCMAKE_INSTALL_PREFIX=/path/to/install ../src
    4) make && make install
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
-ENDIF ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
+ENDIF("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 
 # -----------------------------------------------------------------------------
 # Set up CMake policies (if any)
 # -----------------------------------------------------------------------------
-IF (COMMAND cmake_policy)
-  CMAKE_POLICY(SET CMP0009 NEW)  # Do not follow symlinks w/ FILE_GLOB_RECURSE
+IF(COMMAND cmake_policy)
+  CMAKE_POLICY(SET CMP0009 NEW) # Do not follow symlinks w/ FILE_GLOB_RECURSE
 
-  IF (("${CMAKE_VERSION}" VERSION_GREATER 3.0) OR ("${CMAKE_VERSION}" VERSION_EQUAL 3.0))
-    CMAKE_POLICY(SET CMP0026 OLD)  # Read LOCATION properties from build targets
-    CMAKE_POLICY(SET CMP0042 OLD)  # Disable @rpath in target install name
+  IF(("${CMAKE_VERSION}" VERSION_GREATER 3.0) OR("${CMAKE_VERSION}" VERSION_EQUAL 3.0))
+    CMAKE_POLICY(SET CMP0026 OLD) # Read LOCATION properties from build targets
+    CMAKE_POLICY(SET CMP0042 OLD) # Disable @rpath in target install name
   ENDIF()
-ENDIF (COMMAND cmake_policy)
+ENDIF(COMMAND cmake_policy)
 
 # ------------------------------------------------------------------------------
 # Release information
 # ------------------------------------------------------------------------------
-SET (XCDF_MAJOR_VERSION  3 CACHE STRING "Major version number")
-SET (XCDF_MINOR_VERSION  0 CACHE STRING "Minor version number")
-SET (XCDF_PATCH_VERSION  1 CACHE STRING "Patch number")
+SET(XCDF_MAJOR_VERSION 4 CACHE STRING "Major version number")
+SET(XCDF_MINOR_VERSION 0 CACHE STRING "Minor version number")
+SET(XCDF_PATCH_VERSION 0 CACHE STRING "Patch number")
```

Comment:
The new version should be 3.1.0, not 4.0.0.

Reply:
The python interface has incompatible changes.

We could decouple the python interface Version from the cpp library Version.

Reply:
The major version needs to be bumped up if the versioning refers to the whole package, as the Python API will cause a breaking change with respect to v3.00.03.

If decoupling is fine for @jimbraun, it's fine for me.
Personally, I think it would increase maintenance, and that going for v4.0.0 altogether is easier.

Reply:
Yes, you are right, let's bump the major version to 4.0.0.